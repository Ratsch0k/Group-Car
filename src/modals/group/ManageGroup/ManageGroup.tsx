import React, {useEffect} from 'react';
import {
  RestError,
  useStateIfMounted,
  CenteredCircularProgress,
  useGroups,
  GroupWithOwnerAndMembersAndInvites,
  useApi,
} from 'lib';
import ManageGroupErrorHandler from './ManageGroupNoGroupError';
import {ManageGroupOverview} from './ManageGroupOverview';
import {useParams} from 'react-router-dom';

/**
 * Props for the manage group component.
 */
export interface ManageGroupProps {
  /**
   * Id of the group. If not provided this
   * component will try to get the parameter `:groupId` from
   * the path. If that's not possible it will show an error
   * message.
   */
  groupId?: number;
}

/**
 * Component for managing the specified group.
 * @param props - The props.
 */
export const ManageGroup: React.FC<ManageGroupProps> =
(props: ManageGroupProps) => {
  const {getGroup} = useGroups();
  const {getInvites, getMembers} = useApi();
  const {groupId: groupIdParam} = useParams<{groupId: string}>();
  const [groupData, setGroupData] =
      useStateIfMounted<GroupWithOwnerAndMembersAndInvites | null>(null);
  const [error, setError] = useStateIfMounted<RestError | null | boolean>(null);


  // Get the group
  useEffect(() => {
    // Get the group id
    let selectedGroupId: number;

    if (typeof props.groupId === 'number') {
      selectedGroupId = props.groupId;
    } else {
      // Try to get the groupId from the path
      selectedGroupId = parseInt(groupIdParam);
    }

    if (typeof selectedGroupId !== 'undefined' && !isNaN(selectedGroupId)) {
      // Get group, members and invites
      Promise.all([
        getGroup(selectedGroupId),
        getMembers(selectedGroupId),
        getInvites(selectedGroupId),
      ]).then(([group, members, invites]) => {
        setGroupData({
          ...group.data,
          invites: invites.data.invites,
          members: members.data.members,
        });
      }).catch(() => {
        setError(true);
      });
    } else {
      setError(true);
    }

    // eslint-disable-next-line
  }, [props, groupIdParam]);

  if (groupData === null && error === null) {
    return <CenteredCircularProgress />;
  } else if (error === null && groupData !== null) {
    return <ManageGroupOverview group={groupData}/>;
  } else {
    return <ManageGroupErrorHandler/>;
  }
};

export default ManageGroup;
