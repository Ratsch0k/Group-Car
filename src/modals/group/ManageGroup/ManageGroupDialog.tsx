import React from 'react';
import {
  AutoFullscreenDialog,
  CloseableDialogTitle,
  useModalRouter,
} from 'lib';
import {useTranslation} from 'react-i18next';
import {
  createStyles,
  DialogContent,
  makeStyles,
  Theme,
} from '@material-ui/core';
import ManageGroup from './ManageGroup';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    content: {
      paddingBottom: theme.spacing(3),
    },
  }),
);

export const ManageGroupDialog: React.FC = () => {
  const {t} = useTranslation();
  const {close} = useModalRouter();
  const classes = useStyles();

  return (
    <AutoFullscreenDialog open={true} breakpoint='sm' maxWidth='sm' fullWidth>
      <CloseableDialogTitle close={close}>
        {t('modals.group.manage.title')}
      </CloseableDialogTitle>
      <DialogContent className={classes.content}>
        <ManageGroup />
      </DialogContent>
    </AutoFullscreenDialog>
  );
};

export default ManageGroupDialog;
