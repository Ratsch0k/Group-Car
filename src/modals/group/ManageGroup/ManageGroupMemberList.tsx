import {List} from '@material-ui/core';
import {AuthContext, GroupWithOwnerAndMembers} from 'lib';
import React, {useContext} from 'react';
import ManageGroupMemberListItem from './ManageGroupMemberListItem';

export interface ManageGroupMemberListProps {
  group: GroupWithOwnerAndMembers;
}

export const ManageGroupMemberList: React.FC<ManageGroupMemberListProps> =
(props: ManageGroupMemberListProps) => {
  const {user} = useContext(AuthContext);

  return (
    <List>
      {
        user &&
        <ManageGroupMemberListItem
          key={`member-${user?.id}`}
          // User should always be in members list
          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
          memberData={props.group.members.find((member) =>
            member.User.id === user.id)!}
          last={props.group.members.length === 1}
          isCurrentUser={true}
          isOwner={props.group.ownerId === user.id}
        />
      }
      {props.group.members
          .filter((member) => member.User.id !== user?.id)
          .map((member, index) => {
            return (
              <ManageGroupMemberListItem
                key={`member-${member.User.id}`}
                memberData={member}
                isOwner={props.group.ownerId === member.User.id}
                last={index === props.group.members.length - 1}
              />
            );
          })}
    </List>
  );
};

export default ManageGroupMemberList;
