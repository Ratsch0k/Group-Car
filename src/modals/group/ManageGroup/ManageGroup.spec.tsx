import { render, waitFor } from "@testing-library/react";
import ManageGroup from "./ManageGroup";
import React from "react";
import { Api, ApiContext, theme, GroupContext} from 'lib';
import { MemoryRouter, Route } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core";

it('renders without crashing', () => {
  render(
    <MemoryRouter>
      <ManageGroup groupId={2} />
    </MemoryRouter>
  );
});

it('renders circular progress while loading group data',() => {
  const fakeApi = {
    getGroup: jest.fn().mockRejectedValue(undefined),
  }

  const screen = render (
    <ThemeProvider theme={theme}>
      <MemoryRouter>
        <GroupContext.Provider value={fakeApi as unknown as Api}>
          <ManageGroup groupId={1}/>
        </GroupContext.Provider>
      </MemoryRouter>
    </ThemeProvider>
  );

  expect(screen.baseElement).toMatchSnapshot();
})

it('renders error message if group doesn\'t exist', async () => {
  const fakeApi = {
    getGroup: jest.fn().mockRejectedValue(undefined),
  }

  const screen = render (
    <ThemeProvider theme={theme}>
      <MemoryRouter>
        <GroupContext.Provider value={fakeApi as unknown as Api}>
          <ManageGroup groupId={1}/>
        </GroupContext.Provider>
      </MemoryRouter>
    </ThemeProvider>
  );

  await waitFor(() => expect(fakeApi.getGroup).toHaveBeenCalledTimes(1));
  expect(fakeApi.getGroup).toHaveBeenLastCalledWith(1);
  expect(screen.queryByText('modals.group.manage.loadingFailed')).not.toBeNull();

  expect(screen.baseElement).toMatchSnapshot();  
})

it('renders group name and description if group exists', async () => {
  const fakeGroup = {
    id: 2,
    name: 'fake group name',
    description: 'fake group description',
  };
  const fakeApi = {
    getGroup: jest.fn().mockResolvedValue({data: fakeGroup}),
  }

  const screen = render (
    <ThemeProvider theme={theme}>
      <MemoryRouter>
        <GroupContext.Provider value={fakeApi as unknown as Api}>
            <ManageGroup groupId={2}/>
        </GroupContext.Provider>
      </MemoryRouter>
    </ThemeProvider>
  );

  await waitFor(() => expect(fakeApi.getGroup).toHaveBeenCalledTimes(1));
  expect(fakeApi.getGroup).toHaveBeenLastCalledWith(fakeGroup.id);

  expect(screen.baseElement).toMatchSnapshot();

  
  // Check if name and description exist
  expect(screen.queryByText(fakeGroup.name)).not.toBeNull();
  expect(screen.queryByText(fakeGroup.description)).not.toBeNull();
})

it('get groupId from route params if not provided as property', async () => {
  const fakeGroup = {
    id: 2,
    name: 'name',
    description: 'description',
  };
  const fakeApi = {
    getGroup: jest.fn().mockResolvedValue({data: fakeGroup}),
  }

  const screen = render (
    <ThemeProvider theme={theme}>
      <MemoryRouter initialEntries={[`/group/manage/${fakeGroup.id}`]}>
        <GroupContext.Provider value={fakeApi as unknown as Api}>
          <Route path='/group/manage/:groupId'>
            <ManageGroup />
          </Route>
        </GroupContext.Provider>
      </MemoryRouter>
    </ThemeProvider>
  );

  await waitFor(() => expect(fakeApi.getGroup).toHaveBeenCalledTimes(1));
  expect(fakeApi.getGroup).toHaveBeenLastCalledWith(fakeGroup.id);

  expect(screen.baseElement).toMatchSnapshot();
})
