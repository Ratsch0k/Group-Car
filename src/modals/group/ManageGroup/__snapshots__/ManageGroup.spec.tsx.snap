// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get groupId from route params if not provided as property 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-132"
    >
      <div
        class="makeStyles-overview-133"
      >
        <div
          class="MuiBox-root MuiBox-root-137"
        >
          <h5
            class="MuiTypography-root makeStyles-text-135 MuiTypography-h5 MuiTypography-gutterBottom"
          >
            <span>
              fake group name
            </span>
          </h5>
          <p
            class="MuiTypography-root makeStyles-text-135 makeStyles-description-136 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            <span>
              fake group description
            </span>
          </p>
        </div>
      </div>
      <div
        class="makeStyles-tabsDesktop-134"
        style="height: 400px;"
      >
        <div
          class="MuiPaper-root makeStyles-container-140 MuiPaper-outlined makeStyles-paperOutlined-138 MuiPaper-rounded"
        >
          <div
            class="MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed"
              style="overflow: hidden;"
            >
              <div
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="group-tabpanel-members"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected MuiTab-fullWidth"
                  id="group-tab-members"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    modals.group.manage.tabs.members.title
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="group-tabpanel-cars"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-disabled MuiTab-fullWidth Mui-disabled"
                  disabled=""
                  id="group-tab-cars"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    modals.group.manage.tabs.cars.title
                  </span>
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-141 PrivateTabIndicator-colorSecondary-143 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            class="makeStyles-tabContent-139"
            style="overflow-x: hidden;"
          >
            <div
              class="react-swipeable-view-container"
              style="height: 100%; flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform; transform: translate(0%, 0);"
            >
              <div
                aria-hidden="false"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  aria-labelledby="group-tab-members"
                  class="makeStyles-tabContent-139"
                  id="group-tabpanel-members"
                  role="tabpanel"
                >
                  <ul
                    class="MuiList-root MuiList-padding"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-gutters"
                    >
                      <div
                        class="MuiListItemAvatar-root"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-148"
                        >
                          <img
                            alt="user-2"
                            class="MuiAvatar-img"
                            src="/api/user/2/profile-pic"
                          />
                        </div>
                      </div>
                      <div
                        class="MuiListItemText-root MuiListItemText-multiline"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1"
                        >
                          TEST
                        </p>
                        <span>
                          <div
                            class="MuiChip-root MuiChip-colorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-147"
                            style="margin-right: 4px;"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.owner
                            </span>
                          </div>
                          <div
                            class="MuiChip-root MuiChip-colorSecondary MuiChip-outlined MuiChip-outlinedSecondary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-147"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.admin
                            </span>
                          </div>
                        </span>
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
              <div
                aria-hidden="true"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  aria-labelledby="group-tab-cars"
                  hidden=""
                  id="group-tabpanel-cars"
                  role="tabpanel"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders circular progress while loading group data 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root MuiBox-root-11 makeStyles-centered-9 makeStyles-container-8"
    >
      <div
        class="MuiBox-root MuiBox-root-12 makeStyles-centered-9"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders error message if group doesn't exist 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root MuiBox-root-18 makeStyles-centered-14"
    >
      <div
        class="MuiBox-root MuiBox-root-19 makeStyles-centered-14"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          modals.group.manage.loadingFailed
        </h6>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders group info if group exists 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-25"
    >
      <div
        class="makeStyles-overview-26"
      >
        <div
          class="MuiBox-root MuiBox-root-30"
        >
          <h5
            class="MuiTypography-root makeStyles-text-28 MuiTypography-h5 MuiTypography-gutterBottom"
          >
            <span>
              fake group name
            </span>
          </h5>
          <p
            class="MuiTypography-root makeStyles-text-28 makeStyles-description-29 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            <span>
              fake group description
            </span>
          </p>
        </div>
      </div>
      <div
        class="makeStyles-tabsDesktop-27"
        style="height: 400px;"
      >
        <div
          class="MuiPaper-root makeStyles-container-33 MuiPaper-outlined makeStyles-paperOutlined-31 MuiPaper-rounded"
        >
          <div
            class="MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed"
              style="overflow: hidden;"
            >
              <div
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="group-tabpanel-members"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected MuiTab-fullWidth"
                  id="group-tab-members"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    modals.group.manage.tabs.members.title
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="group-tabpanel-cars"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-disabled MuiTab-fullWidth Mui-disabled"
                  disabled=""
                  id="group-tab-cars"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    modals.group.manage.tabs.cars.title
                  </span>
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-34 PrivateTabIndicator-colorSecondary-36 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            class="makeStyles-tabContent-32"
            style="overflow-x: hidden;"
          >
            <div
              class="react-swipeable-view-container"
              style="height: 100%; flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform; transform: translate(0%, 0);"
            >
              <div
                aria-hidden="false"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  aria-labelledby="group-tab-members"
                  class="makeStyles-tabContent-32"
                  id="group-tabpanel-members"
                  role="tabpanel"
                >
                  <ul
                    class="MuiList-root MuiList-padding"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-gutters"
                    >
                      <div
                        class="MuiListItemAvatar-root"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-51"
                        >
                          <img
                            alt="user-2"
                            class="MuiAvatar-img"
                            src="/api/user/2/profile-pic"
                          />
                        </div>
                      </div>
                      <div
                        class="MuiListItemText-root MuiListItemText-multiline"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1"
                        >
                          misc.you
                        </p>
                        <span>
                          <div
                            class="MuiChip-root MuiChip-colorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-40"
                            style="margin-right: 4px;"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.owner
                            </span>
                          </div>
                          <div
                            class="MuiChip-root MuiChip-colorSecondary MuiChip-outlined MuiChip-outlinedSecondary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-40"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.admin
                            </span>
                          </div>
                        </span>
                      </div>
                    </li>
                  </ul>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiFab-root makeStyles-fab-41 MuiFab-secondary"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiFab-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  aria-labelledby="group-tab-cars"
                  hidden=""
                  id="group-tabpanel-cars"
                  role="tabpanel"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders list of invites correctly 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-99"
    >
      <div
        class="makeStyles-overview-100"
      >
        <div
          class="MuiBox-root MuiBox-root-104"
        >
          <h5
            class="MuiTypography-root makeStyles-text-102 MuiTypography-h5 MuiTypography-gutterBottom"
          >
            <span>
              fake group name
            </span>
          </h5>
          <p
            class="MuiTypography-root makeStyles-text-102 makeStyles-description-103 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            <span>
              fake group description
            </span>
          </p>
        </div>
      </div>
      <div
        class="makeStyles-tabsDesktop-101"
        style="height: 400px;"
      >
        <div
          class="MuiPaper-root makeStyles-container-107 MuiPaper-outlined makeStyles-paperOutlined-105 MuiPaper-rounded"
        >
          <div
            class="MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed"
              style="overflow: hidden;"
            >
              <div
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="group-tabpanel-members"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected MuiTab-fullWidth"
                  id="group-tab-members"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    modals.group.manage.tabs.members.title
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="group-tabpanel-cars"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-disabled MuiTab-fullWidth Mui-disabled"
                  disabled=""
                  id="group-tab-cars"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    modals.group.manage.tabs.cars.title
                  </span>
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-108 PrivateTabIndicator-colorSecondary-110 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            class="makeStyles-tabContent-106"
            style="overflow-x: hidden;"
          >
            <div
              class="react-swipeable-view-container"
              style="height: 100%; flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform; transform: translate(0%, 0);"
            >
              <div
                aria-hidden="false"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  aria-labelledby="group-tab-members"
                  class="makeStyles-tabContent-106"
                  id="group-tabpanel-members"
                  role="tabpanel"
                >
                  <ul
                    class="MuiList-root MuiList-padding"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
                    >
                      <div
                        class="MuiListItemAvatar-root"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-125"
                        >
                          <img
                            alt="user-2"
                            class="MuiAvatar-img"
                            src="/api/user/2/profile-pic"
                          />
                        </div>
                      </div>
                      <div
                        class="MuiListItemText-root MuiListItemText-multiline"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1"
                        >
                          misc.you
                        </p>
                        <span>
                          <div
                            class="MuiChip-root MuiChip-colorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-114"
                            style="margin-right: 4px;"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.owner
                            </span>
                          </div>
                          <div
                            class="MuiChip-root MuiChip-colorSecondary MuiChip-outlined MuiChip-outlinedSecondary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-114"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.admin
                            </span>
                          </div>
                        </span>
                      </div>
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
                    >
                      <div
                        class="MuiListItemText-root MuiListItemText-inset MuiListItemText-multiline"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                        >
                          ADMIN_1
                        </p>
                        <p
                          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                        >
                          misc.invitedBy
                        </p>
                      </div>
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-gutters"
                    >
                      <div
                        class="MuiListItemText-root MuiListItemText-inset MuiListItemText-multiline"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                        >
                          ADMIN_2
                        </p>
                        <p
                          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                        >
                          misc.invitedBy
                        </p>
                      </div>
                    </li>
                  </ul>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiFab-root makeStyles-fab-115 MuiFab-secondary"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiFab-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  aria-labelledby="group-tab-cars"
                  hidden=""
                  id="group-tabpanel-cars"
                  role="tabpanel"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders list of members correctly 1`] = `
<body>
  <div>
    <div
      class="makeStyles-container-58"
    >
      <div
        class="makeStyles-overview-59"
      >
        <div
          class="MuiBox-root MuiBox-root-63"
        >
          <h5
            class="MuiTypography-root makeStyles-text-61 MuiTypography-h5 MuiTypography-gutterBottom"
          >
            <span>
              fake group name
            </span>
          </h5>
          <p
            class="MuiTypography-root makeStyles-text-61 makeStyles-description-62 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            <span>
              fake group description
            </span>
          </p>
        </div>
      </div>
      <div
        class="makeStyles-tabsDesktop-60"
        style="height: 400px;"
      >
        <div
          class="MuiPaper-root makeStyles-container-66 MuiPaper-outlined makeStyles-paperOutlined-64 MuiPaper-rounded"
        >
          <div
            class="MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed"
              style="overflow: hidden;"
            >
              <div
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="group-tabpanel-members"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected MuiTab-fullWidth"
                  id="group-tab-members"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    modals.group.manage.tabs.members.title
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="group-tabpanel-cars"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-disabled MuiTab-fullWidth Mui-disabled"
                  disabled=""
                  id="group-tab-cars"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    modals.group.manage.tabs.cars.title
                  </span>
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-67 PrivateTabIndicator-colorSecondary-69 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            class="makeStyles-tabContent-65"
            style="overflow-x: hidden;"
          >
            <div
              class="react-swipeable-view-container"
              style="height: 100%; flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform; transform: translate(0%, 0);"
            >
              <div
                aria-hidden="false"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  aria-labelledby="group-tab-members"
                  class="makeStyles-tabContent-65"
                  id="group-tabpanel-members"
                  role="tabpanel"
                >
                  <ul
                    class="MuiList-root MuiList-padding"
                  >
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
                    >
                      <div
                        class="MuiListItemAvatar-root"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-88"
                        >
                          <img
                            alt="user-2"
                            class="MuiAvatar-img"
                            src="/api/user/2/profile-pic"
                          />
                        </div>
                      </div>
                      <div
                        class="MuiListItemText-root MuiListItemText-multiline"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1"
                        >
                          misc.you
                        </p>
                        <span>
                          <div
                            class="MuiChip-root MuiChip-colorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-73"
                            style="margin-right: 4px;"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.owner
                            </span>
                          </div>
                          <div
                            class="MuiChip-root MuiChip-colorSecondary MuiChip-outlined MuiChip-outlinedSecondary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-73"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.admin
                            </span>
                          </div>
                        </span>
                      </div>
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
                    >
                      <div
                        class="MuiListItemAvatar-root"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-90"
                        >
                          <img
                            alt="user-12"
                            class="MuiAvatar-img"
                            src="/api/user/12/profile-pic"
                          />
                        </div>
                      </div>
                      <div
                        class="MuiListItemText-root MuiListItemText-multiline"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1"
                        >
                          ADMIN
                        </p>
                        <span>
                          <div
                            class="MuiChip-root MuiChip-colorSecondary MuiChip-outlined MuiChip-outlinedSecondary MuiChip-sizeSmall WithStyles(ForwardRef(Chip))-sizeSmall-73"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall"
                            >
                              misc.admin
                            </span>
                          </div>
                        </span>
                      </div>
                    </li>
                    <li
                      class="MuiListItem-root MuiListItem-gutters MuiListItem-divider"
                    >
                      <div
                        class="MuiListItemAvatar-root"
                      >
                        <div
                          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-92"
                        >
                          <img
                            alt="user-13"
                            class="MuiAvatar-img"
                            src="/api/user/13/profile-pic"
                          />
                        </div>
                      </div>
                      <div
                        class="MuiListItemText-root MuiListItemText-multiline"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1"
                        >
                          MEMBER
                        </p>
                        <span />
                      </div>
                    </li>
                  </ul>
                  <div
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiFab-root makeStyles-fab-78 MuiFab-secondary"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="MuiFab-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </div>
                </div>
              </div>
              <div
                aria-hidden="true"
                data-swipeable="true"
                style="width: 100%; flex-shrink: 0; overflow: auto;"
              >
                <div
                  aria-labelledby="group-tab-cars"
                  hidden=""
                  id="group-tabpanel-cars"
                  role="tabpanel"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
